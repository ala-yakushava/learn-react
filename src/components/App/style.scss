@import "../../styles/theme";

.App {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background-color: $darkest-grey;
  color: $white;

  &_space {
    height: 10px;
    background-color: $grey;
  }
}
