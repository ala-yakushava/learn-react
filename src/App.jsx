import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { hot } from 'react-hot-loader';

import './style.scss';
import RouterComponent from './routes';

const App = ({
  Router, location, store,
}) => (
  <React.StrictMode>
    <Provider store={store}>
      <Router location={location}>
        <RouterComponent />
      </Router>
    </Provider>
  </React.StrictMode>
);

App.propTypes = {
  Router: PropTypes.func.isRequired,
  location: PropTypes.string,
  store: PropTypes.shape({
    dispatch: PropTypes.func.isRequired,
    getState: PropTypes.func.isRequired,
  }).isRequired,
};
App.defaultProps = {
  location: null,
};

export default hot(module)(App);
