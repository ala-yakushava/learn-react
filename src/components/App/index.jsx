import React from 'react';

import './style.scss';
import Container from '../Container';
import Header from '../Header';
import MovieCatalog from '../MovieCatalog';
import Footer from '../Footer';

const App = () => {
  return (
    <div className="App">
      <Container>
        <Header />
      </Container>
      <div className="App_space"></div>
      <Container>
        <MovieCatalog />
      </Container>
      <Footer />
    </div>
  );
}

export default App;
