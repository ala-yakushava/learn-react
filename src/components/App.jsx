import React from 'react';
import './App.scss';

const App = () => {
  return (
    <div className="App">
      <header>
        Hello World!
      </header>
    </div>
  );
}

export default App;
