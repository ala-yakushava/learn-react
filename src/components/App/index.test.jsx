import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import App from './index';

jest.mock('react-redux');

describe('Components - App', () => {
  const dispatch = jest.fn();

  const state = {
    moviesInfo: { movies: [] },
    suggestion: { filterValue: 'All', sortValue: 'release_date' },
  };

  beforeAll(() => {
    useSelector.mockImplementation((cb) => cb(state));
    useDispatch.mockReturnValue(dispatch);
  });

  afterAll(() => {
    useSelector.mockClear();
    useDispatch.mockClear();
  });

  test('should match snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={['/movies']}>
          <App>
            <div>Content</div>
          </App>
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
