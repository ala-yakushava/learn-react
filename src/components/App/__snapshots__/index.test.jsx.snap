// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - App should match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="Container"
  >
    <div>
      Content
    </div>
  </div>
  <div
    className="App_space"
  />
  <div
    className="Container"
  >
    <div
      className="MovieCatalog"
      data-testid="catalog"
    >
      <div
        className="MovieCatalog_filter"
      >
        <div
          className="Filter"
          data-testid="filter"
        >
          <button
            className="Filter_item active"
            onClick={[Function]}
            type="button"
          >
            All
          </button>
          <button
            className="Filter_item"
            onClick={[Function]}
            type="button"
          >
            Documentary
          </button>
          <button
            className="Filter_item"
            onClick={[Function]}
            type="button"
          >
            Comedy
          </button>
          <button
            className="Filter_item"
            onClick={[Function]}
            type="button"
          >
            Horror
          </button>
          <button
            className="Filter_item"
            onClick={[Function]}
            type="button"
          >
            Crime
          </button>
        </div>
        <div
          className="Sorting"
          data-testid="sorting"
        >
          <span
            className="Sorting_label"
          >
            Sort by
          </span>
          <select
            className="Select"
            onChange={[Function]}
            value="release_date"
          >
            <option
              value="vote_average"
            >
              Rating
            </option>
            <option
              value="release_date"
            >
              Release date
            </option>
          </select>
        </div>
      </div>
      <section
        className="MovieNotFound"
        data-testid="movie-not-found"
      >
        <div
          className="Heading"
        >
          No Movie Found
        </div>
      </section>
    </div>
  </div>
  <footer
    className="Footer"
  >
    <p
      className="Logo"
    >
      <b>
        netflix
      </b>
      roulette
    </p>
  </footer>
</div>
`;
